bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/DateExpression.java

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.common;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.TimeZone;
+
+import org.eclipse.smarthome.core.common.DateExpression.DateExpressionPart;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <code>DateExpression</code> is an implementation of {@link Expression} that provides a parser and evaluator for for
+ * ISO8601 date expressions (https://en.wikipedia.org/wiki/ISO_8601). ISO8601 expressions provide the ability to specify
+ * simple yet precise dates.
+ * <P>
+ * ISO8601 expressions are comprised of 7 required fields described as follows:
+ *
+ * YYYY-MM-DDThh:mm:ss.sTZD
+ *
+ * where
+ *
+ * YYYY = four-digit year
+ * MM = two-digit month (01=January, etc.)
+ * DD = two-digit day of month (01 through 31)
+ * hh = two digits of hour (00 through 23) (am/pm NOT allowed)
+ * mm = two digits of minute (00 through 59)
+ * ss = two digits of second (00 through 59)
+ * s = one or more digits representing a decimal fraction of a second
+ * TZD = time zone designator (Z or +hh:mm or -hh:mm)
+ *
+ * IMPORTANT NOTE : ISO8601 expressions that specify fractions of seconds are NOT supported
+ *
+ * @author Karel Goderis - Intial Contribution
+ *
+ */
+public class DateExpression extends AbstractExpression<DateExpressionPart> {
+
+    private final static Logger logger = LoggerFactory.getLogger(DateExpression.class);
```

Comment:
As already stated, you should use a non-static logger instance if possible. In this case, we do not need the logger instance at all. Nothing is logged in the current code base.